core/mount/temp.go

```diff
@@ -67,6 +69,47 @@ func WithTempMount(ctx context.Context, mounts []Mount, f func(root string) erro
 	return nil
 }
 
+// removeVolatileTempMount The volatile option of overlayfs doesn't allow to mount again using the
+// same upper / work dirs. Since it's a temp mount, avoid using that
+// option here. Reference: https://docs.kernel.org/filesystems/overlayfs.html#volatile-mount
+// TODO: Make this logic conditional once the kernel supports reusing
+// overlayfs volatile mounts.
+func removeVolatileTempMount(mounts []Mount) []Mount {
+	var out []Mount
+	for i, m := range mounts {
+		if m.Type != "overlay" {
+			continue
+		}
+		for j, opt := range m.Options {
+			if opt == "volatile" {
+				if out == nil {
+					out = copyMounts(mounts)
+				}
+				out[i].Options[j] = ""
```

Comment:
One more optimization here, don't have copyMounts copy the Options slice. Just copy the option slice here when removing the element. Its fine to keep the original slice array until it is modified.

out[i].Options[j] = append(out[i].Options[:j], out[i].Options[j+1:]...)

Reply:
Great idea! Made those adjustments. Really appreciate the help you are providing!